### ISSUE TITLE
SVG input with solid black renders to resized bitmap with close-to-but-not-exactly black

### ISSUE THREAD

**Issue by hfhchan** [lovell/sharp#2968]
Are you using the latest version? `0.29.2`

What are the steps to reproduce?
1. Export an SVG with await sharp(fs.readFileSync(inputPath)).resize(128, 128).toFile(outputPath);
2. Result: the pixels exported fluctuate between rgb(0,0,0) and rgb(4,4,4)

What is the expected behaviour?
The center part of the shape should be solid rgb(0,0,0)

Are you able to provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem?

`file.svg`
````
<?xml version="1.0" encoding="utf-8"?>
<svg width="1024" height="1080" xmlns="http://www.w3.org/2000/svg">
<path fill="black" d="M 141 195 L 302 195 L 335 169 L 394 217 L 367 238 L 367 899 L 302 931 L 302 867 L 141 867 L 141 931 L 76 964 L 76 163 M 608 937 L 781 937 L 781 560 L 667 560 L 667 539 L 781 539 L 781 195 L 641 195 L 641 173 L 876 173 L 930 120 L 994 173 L 994 195 L 846 195 L 846 539 L 865 539 L 918 485 L 983 539 L 983 560 L 846 560 L 846 937 L 887 937 L 940 883 L 1005 937 L 1005 958 L 608 958 M 581 966 L 573 977 L 564 986 L 553 994 L 542 1001 L 530 1006 L 517 1009 L 504 1011 L 490 1012 L 490 980 L 382 964 L 382 948 L 490 948 L 497 947 L 503 946 L 509 944 L 514 943 L 518 940 L 522 937 L 525 934 L 529 929 L 532 924 L 545 883 L 554 825 L 560 752 L 561 665 L 556 587 L 539 622 L 498 692 L 475 724 L 451 755 L 425 784 L 396 810 L 389 805 L 445 706 L 479 637 L 494 601 L 520 528 L 531 489 L 540 455 L 523 371 L 520 358 L 487 401 L 458 432 L 428 459 L 396 483 L 389 476 L 436 414 L 479 344 L 500 307 L 503 300 L 495 273 L 457 180 L 412 94 L 420 88 L 482 167 L 525 251 L 553 180 L 568 133 L 582 88 L 585 74 L 644 101 L 657 116 L 637 122 L 627 152 L 607 201 L 586 246 L 563 290 L 548 314 L 564 360 L 589 463 L 603 467 L 594 493 L 606 565 L 615 663 L 617 753 L 614 831 L 607 895 L 594 942 L 588 955 M 141 641 L 141 845 L 302 845 L 302 641 M 141 429 L 141 619 L 302 619 L 302 429 M 141 217 L 141 407 L 302 407 L 302 217" />
</svg>
````
`app.js`
````
await sharp(fs.readFileSync("file.svg")).resize(128, 128).toFile("output.png");
````

Are you able to provide a sample image that helps explain the problem?

What is the output of running `npx envinfo --binaries --system`?
````
  System:
    OS: macOS 11.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 51.16 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.0.1 - /usr/local/bin/node
    npm: 8.1.0 - /usr/local/bin/npm
````

**Comment by hfhchan** [comment]
Sorry I accidentally pressed Enter too quickly.  Let me edit back the info.

**Comment by hfhchan** [comment]
I think that this is because sharp is rendering the SVG at its original resolution, and then downsampling the image using an algorithm that takes into account a lot of pixels, rather than directly rendering the SVG to 128x128.

**Comment by lovell** [comment]
Hi, did you see the `density` parameter, which can be used to control the DPI when rendering vector images?

https://sharp.pixelplumbing.com/api-constructor#parameters

```js
await sharp("file.svg", { density: 9 }).toFile("output.png");
```

**Comment by hfhchan** [comment]
Yes I did. Unfortunately some images vary in width and height slightly, so that would mean calculating the density for each image, so I did not take that path.

I took a different path: patch in a viewbox attribute with the width and height, then patch the width and height to the target output resolution so they can be exported at the native resolution automatically.

Many thanks for your time and help.
